### PR TITLE
Feature/x 3 modify meal

### PR DESCRIPTION
**Pull request by algpuentes**

### CODE REVIEW

File: src/main/java/com/xantar/xantarcore/db/EMeal.java [hunk 1]
```diff
@@ -36,4 +40,85 @@
...
+
+	private boolean compareAttributes(Meal meal) {
+		return Objects.equals(this.id, meal.id)
+				&& Objects.equals(this.name, meal.name)
+				&& Objects.equals(this.description, meal.description)
+				&& Objects.equals(this.slots, meal.slots)
+				&& Objects.equals(this.imageThumb, meal.imageThumb);
+	}
+
+	public static class EMealBuilder {
```
Comment (fabioir) [inline src/main/java/com/xantar/xantarcore/db/EMeal.java:81]: 👏

File: src/main/java/com/xantar/xantarcore/db/EMeal.java [hunk 2]
```diff
@@ -36,4 +40,85 @@
...
+	}
+
+	public void addSlot(ESlot slot) {
+		this.slots.add(slot);
+		slot.meals.add(this);
+	}
+
+	public void removeSlot(ESlot tag) {
+		this.slots.remove(tag);
+		tag.meals.remove(this);
```
Comment (fabioir) [inline src/main/java/com/xantar/xantarcore/db/EMeal.java]:
```suggestion
		slot.meals.remove(this);
```

File: src/main/java/com/xantar/xantarcore/db/EMeal.java [hunk 3]
```diff
@@ -36,4 +40,85 @@
...
+	@Override
+	public int hashCode() {
+		return Objects.hash(this.id, this.name, this.description, this.slots, this.imageThumb);
+	}
+
+	@Override
+	public boolean equals(Object other) {
+		return Optional.ofNullable(other)
+				.filter(Meal.class::isInstance)
+				.map(Meal.class::cast)
```
Comment (fabioir) [inline src/main/java/com/xantar/xantarcore/db/EMeal.java]:
```suggestion
				.filter(EMeal.class::isInstance)
				.map(EMeal.class::cast)
```

File: src/main/java/com/xantar/xantarcore/db/EMeal.java [hunk 4]
```diff
@@ -36,4 +40,85 @@
...
+				.filter(object -> this.compareAttributes(object))
+				.isPresent();
+	}
+
+	private boolean compareAttributes(Meal meal) {
```
Comment (fabioir) [inline src/main/java/com/xantar/xantarcore/db/EMeal.java]:
```suggestion
	private boolean compareAttributes(EMeal meal) {
```

File: src/main/java/com/xantar/xantarcore/db/ESlot.java [hunk 5]
```diff
@@ -12,6 +20,38 @@
...
+	public void removeSlot(EMeal meal) {
+		this.meals.remove(meal);
+		meal.slots.remove(this);
+	}
+
+	@Override
+	public boolean equals(Object other) {
+		return Optional.ofNullable(other)
+				.filter(Slot.class::isInstance)
+				.map(Slot.class::cast)
```
Comment (fabioir) [inline src/main/java/com/xantar/xantarcore/db/ESlot.java]:
```suggestion
				.filter(ESlot.class::isInstance)
				.map(ESlot.class::cast)
```

File: src/main/java/com/xantar/xantarcore/db/ESlot.java [hunk 6]
```diff
@@ -12,6 +20,38 @@
...
+				.filter(object -> this.compareAttributes(object))
+				.isPresent();
+	}
+
+	private boolean compareAttributes(Slot slot) {
```
Comment (fabioir) [inline src/main/java/com/xantar/xantarcore/db/ESlot.java]:
```suggestion
	private boolean compareAttributes(ESlot slot) {
```

File: src/main/java/com/xantar/xantarcore/db/MealsService.java [hunk 7]
```diff
@@ -40,6 +42,37 @@ public Meal createMeal(Meal meal) {
...
+		return EMealMapper.toModel(this.mealsRepository.save(updateEMeal));
+	}
+
+	/**
+	 * Updates just non null values
+	 * **/
+	private EMeal updateEntity(EMeal originalEMeal, Meal updatedData) {
+		final EMeal eMeal =  new EMeal.EMealBuilder()
+				.withId(originalEMeal.id)
+				.withName(updatedData.name != null ? updatedData.name : originalEMeal.name)
```
Comment (fabioir) [inline src/main/java/com/xantar/xantarcore/db/MealsService.java]: What if the name is empty? ('') 
Should this be allowed?

File: src/main/java/com/xantar/xantarcore/db/MealsService.java [hunk 8]
```diff
@@ -40,6 +42,37 @@ public Meal createMeal(Meal meal) {
...
+				.withDescription(updatedData.description  != null ? updatedData.description : originalEMeal.description)
+				.withImageThumb(updatedData.imageThumb != null ? updatedData.imageThumb : originalEMeal.imageThumb)
+				.build();
```
Comment (fabioir) [inline src/main/java/com/xantar/xantarcore/db/MealsService.java:79]: We are lacking the properties of "MealDetails"

**Review by fabioir — APPROVED** [review]
We need to decide if we use MealDetails before closing this story

File: src/main/java/com/xantar/xantarcore/db/MealsService.java (hunk 7 above)
Comment (algpuentes) [reply src/main/java/com/xantar/xantarcore/db/MealsService.java]: As discussed, no, it should not be allowed.

File: src/main/java/com/xantar/xantarcore/db/MealsService.java (hunk 8 above)
Comment (algpuentes) [reply src/main/java/com/xantar/xantarcore/db/MealsService.java:79]: As discussed, further properties for Meal will be part of the next iteration.

**Pull request merged**